planner/core/find_best_task.go

```diff
@@ -932,6 +933,10 @@ func (ds *DataSource) isCoveringIndex(columns, indexColumns []*expression.Column
 		if !indexCoveringCol(col, indexColumns, idxColLens) && !indexCoveringCol(col, ds.commonHandleCols, ds.commonHandleLens) {
 			return false
 		}
+		if collate.NewCollationEnabled() && indexCoveringCol(col, ds.commonHandleCols, ds.commonHandleLens) &&
```

Comment:
We should check indexCoveringCol for index.
If the column can bound in both index and common handle, we should return true.